primus.js

```diff
@@ -621,15 +621,29 @@ Primus.prototype.initialise = function initialise(options) {
     }
   }
 
+  primus.addGlobalListeners();
+
+  return primus;
+};
+
+Primus.prototype.addGlobalListeners = function () {
   if (window.addEventListener) {
-    window.addEventListener('offline', offline, false);
-    window.addEventListener('online', online, false);
+    window.addEventListener('offline', this.offlineHandler, false);
+    window.addEventListener('online', this.onlineHandler, false);
   } else if (document.body.attachEvent){
-    document.body.attachEvent('onoffline', offline);
-    document.body.attachEvent('ononline', online);
+    document.body.attachEvent('onoffline', this.offlineHandler);
+    document.body.attachEvent('ononline', this.onlineHandler);
   }
+};
 
-  return primus;
+Primus.prototype.removeGlobalListeners = function () {
```

Comment:
Indeed, thanks. I could also move the 2 handler functions inside ```addGlobalListeners``` for readability - let me know if you want me to do this.